numba/np/arrayobj.py

```diff
@@ -5989,17 +5984,15 @@ def get_sort_func(kind, is_float, is_argsort=False):
         return func
```

Comment:
Unrelated to this PR but seems like there is no error being raised here if a key is supplied other than those listed in these branches ?

```python
import numpy as np
import numba

@numba.njit
def foo(x):
    return x.argsort(kind='somesort')

x = np.random.default_rng(1).integers(10, size=10)

print(foo(x))
# Error: func accessed before assignment.
```